File: aoc2022/day1.py
with open('input/day1.in') as f:
    inp = [[int(x) for x in l.split('\n') if x != ''] for l in f.read().split('\n\n')]
    ansP1 = max(map(sum, inp))
    ansP2 = sum(sorted(map(sum, inp), reverse = True)[:3])
print(f'Part 1: {ansP1}')
print(f'Part 2: {ansP2}')

File: aoc2022/day1.hs
import Data.List.Extra
import Functions

main = do
  input <- lines <$> readFile "input/day1.in"
  let nums = map sum (parseGroup input [])
  let ansP1 = maximum nums
  let ansP2 = sum (take 3 (reverse $ sort nums))
  print ansP1
  print ansP2
